File: 3º periodo/POO/18_07/contas/conta.cpp
#include <iostream>
#include <string>

#include "conta.hpp"

Conta::Conta(string descricao, string dataVencimento, string dataPagamento) : 
    descricao(descricao), dataVencimento(dataVencimento), dataPagamento(dataPagamento) {};

Conta::~Conta() {};

string Conta::getDescricao() const {
    return descricao;
}

void Conta::setDescricao(string descricao) {
    this->descricao = descricao;
}

string Conta::getDataVencimento() const {
    return dataVencimento;
}

void Conta::setDataVencimento(string dataVencimento) {
    this->dataVencimento = dataVencimento;
}

string Conta::getDataPagamento() const {
    return dataPagamento;
}

void Conta::setDataPagamento(string dataPagamento) {
    this->dataPagamento = dataPagamento;
}

void fazerPagamentoContas(vector<Conta> *contas, vector<Conta>::iterator *it) {
    Conta aux;
    limpaTela();
    getc(stdin);

    string description, dataVencimento, dataPagamento;
    cout << endl << "Descricao da conta: " << endl;
    getline(cin, description);
    aux.setDescricao(description);// set descrição

    cout << endl << "Data de vencimento: " << endl;
    getline(cin, dataVencimento);
    aux.setDataVencimento(dataVencimento);// set data de vencimento

    cout << endl << "Data de pagamento: " << endl;
    getline(cin, dataPagamento);
    aux.setDataPagamento(dataPagamento);// set data de pagamento

    (*contas).insert((*it), aux);   
}

File: 3º periodo/POO/TP02-Codigo/src/main/java/tools/Conta.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package tools;

public class Conta {
    private String descricao;
    private String dataVencimento;
    private String dataPagamento;
    private Double valor = 50.0;

    public Conta(String descricao, String dataVencimento, String dataPagamento) {
        this.descricao = descricao;
        this.dataVencimento = dataVencimento;
        this.dataPagamento = dataPagamento;
    }

    public String getDescricao() {
        return descricao;
    }

    public void setDescricao(String descricao) {
        this.descricao = descricao;
    }

    public String getDataVencimento() {
        return dataVencimento;
    }

    public void setDataVencimento(String dataVencimento) {
        this.dataVencimento = dataVencimento;
    }

    public String getDataPagamento() {
        return dataPagamento;
    }

    public void setDataPagamento(String dataPagamento) {
        this.dataPagamento = dataPagamento;
    }
    
    public double getValor(){
        return valor;
    }
}